### PR TITLE
Remove error prefix from ray_tracing errors and improve error testing

### LINKED ISSUE #17030: Remove extraneous error information in ray_tracing and improve error testing
## Reason
#16864 added information that already existed in many of the ray_tracing module errors. This information can be removed now.

## Design
Remove the extraneous error information in the ray_tracing module. Add more error testing.

## Impact
Removal of duplicate information in errors. Better coverage.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #17030

### CODE REVIEW

**Comment by loganharbour** [comment]
@GiudGiud if you wanna take a look at this test-heavy behemoth, I wouldn't object :)

File: modules/ray_tracing/test/tests/userobjects/ray_tracing_study/errors/tests [hunk 1]
```diff
@@ -27,7 +27,167 @@
...
+      allow_test_objects = true
+      detail = 'when setting a ray\'s starting direction before setting its starting information,'
+    []
+    [ray_set_direction_again]
+      type = RunException
+      input = 'errors.i'
+      cli_args = 'UserObjects/study/ray_set_direction_again=true'
+      expect_err = 'Cannot change a Ray\'s starting direction using Ray::setStartingDirection\(\)'
+      allow_test_objects = true
+      detail = 'when setting a ray\'s starting direction multiple times without clearning it,'
```
Comment (GiudGiud) [inline modules/ray_tracing/test/tests/userobjects/ray_tracing_study/errors/tests:110]:
```suggestion
      detail = 'when setting a ray\'s starting direction multiple times without clearing it,'
```

**Review by GiudGiud — COMMENTED** [review]
Nothing major. This is really good testing, this module is set to be very resilient.

File: modules/ray_tracing/src/raybcs/RayBoundaryConditionBase.C [hunk 2]
```diff
@@ -55,25 +55,16 @@ RayBoundaryConditionBase::changeRayDirection(const Point & direction, const bool
   auto & ray = currentRay();
 
   if (!ray->shouldContinue())
-    mooseError(_error_prefix,
-               ": Cannot changeRayDirection() for a Ray that should not continue\n\n",
+    mooseError("Cannot changeRayDirection() for a Ray that should not continue\n\n",
                ray->getInfo());
 
-  if (!ray->intersections())
```
[the remark this answers is not in the document]
Comment (loganharbour) [reply modules/ray_tracing/src/raybcs/RayBoundaryConditionBase.C:62]: Yep. This is actually not a requirement.

File: modules/ray_tracing/src/raybcs/RayBoundaryConditionBase.C [hunk 3]
```diff
@@ -99,22 +90,5 @@ RayBoundaryConditionBase::moveRayToBuffer(std::shared_ptr<Ray> & ray)
   mooseAssert(_study.currentlyPropagating(),
               "Should not move Rays into buffer while not propagating");
 
-  if (_current_elem != ray->currentElem())
```
[the remark this answers is not in the document]
Comment (loganharbour) [reply modules/ray_tracing/src/raybcs/RayBoundaryConditionBase.C:102]: This is actually not a requirement. The setting of the information _on_ the ray will do the right validation checking that is needed. There's a specific case here: you could hit a boundary at a node and want the ray to start on another element at the node.

File: modules/ray_tracing/src/userobjects/RayTracingStudy.C [hunk 4]
```diff
@@ -170,7 +171,6 @@ RayTracingStudy::RayTracingStudy(const InputParameters & parameters)
 
     _called_initial_setup(false),
     _need_to_associate_registered_rays(false),
-    _verify_ray_error_prefix("While moving Ray to buffer in " + _error_prefix),
```
[the remark this answers is not in the document]
Comment (loganharbour) [reply modules/ray_tracing/src/userobjects/RayTracingStudy.C:173]: Yep. You'll find `errorPrefix()` a few times called on the study.

**Comment by loganharbour** [comment]
Good to go again. Thanks for looking through this.

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**